### PR TITLE
include draintimeout and change node drain method.

### PR DESCRIPTION
**Pull request by shreyas-badiger**
- [x] Include the `timeout` flag in the drain command.
- [x] Replace the kubectl-binary drain command with kubectl-library drain function.
- [x] Update packages and dependencies.
- [x] Testing
    - [x] functional testing
    - [x] unit tests
    - [x] test with 100+ nodes
    - [x] edge cases - test with blocking PDBs and other scenarios.

### CODE REVIEW

File: pkg/service/nodes.go [hunk 1]
```diff
@@ -75,18 +76,23 @@ func runCommandWithContext(call string, args []string, timeoutSeconds, retryInte
 	defer cancel()
 	err := retry.Do(
 		func() error {
+			log.Info("[sbadiger] about to excute command: ", call, " args: ", args)
 			cmd := exec.CommandContext(ctx, call, args...)
 			_, err := cmd.CombinedOutput()
 			if err != nil {
+				log.Info("[sbadiger] command execution failed: ", call, " args: ", args, " error: ", err)
```
Comment (ZihanJiang96) [inline pkg/service/nodes.go:83]: clean the log info please

Comment (shreyas-badiger) [reply pkg/service/nodes.go:83]: lol. Missed it. Thanks.

**Comment by shreyas-badiger** [comment]
Created the PR from the wrong branch. Converting it to Draft again.

File: pkg/service/nodes.go (hunk 1 above)
Comment (shreyas-badiger) [reply pkg/service/nodes.go:83]: Looks like I created the PR from the wrong branch. I will collapse the branch and open the PR for review.

File: pkg/service/nodes.go [hunk 2]
```diff
@@ -49,51 +53,20 @@ func isNodeStatusInCondition(node v1.Node, condition v1.ConditionStatus) bool {
...
-	err := runCommandWithContext(drainCommand, drainArgs, timeout, retryInterval, retryAttempts)
-	if err != nil {
-		if err.Error() == "command execution timed out" {
-			log.Warnf("failed to drain node %v, drain command timed-out", nodeName)
+	for retryAttempts > 0 {
+		err := drainNodeUtil(node, int(timeout), kubeClient)
+		if err != nil {
+			log.Errorf("failed to drain node %v  error: %v ", node.Name, err)
 			return err
 		}
```
Comment (ZihanJiang96) [inline pkg/service/nodes.go:67]: if err is nil, should we return directly?

File: pkg/service/events_test.go [hunk 3]
```diff
@@ -52,7 +53,7 @@ func Test_PublishEvent(t *testing.T) {
 	publishKubernetesEvent(kubeClient, event)
 	expectedEvents := 1
 
-	events, err := kubeClient.CoreV1().Events(EventNamespace).List(metav1.ListOptions{})
+	events, err := kubeClient.CoreV1().Events(EventNamespace).List(context.Background(), metav1.ListOptions{})
```
Comment (akhil-rane) [inline pkg/service/events_test.go:56]: Just curious, how does adding context help here?

File: pkg/service/nodes.go [hunk 4]
```diff
@@ -3,20 +3,24 @@ package service
...
 	"fmt"
+	"os"
 	"os/exec"
 	"strings"
 	"time"
 
-	"github.com/avast/retry-go"
-	"github.com/keikoproj/lifecycle-manager/pkg/log"
+	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/api/core/v1"
```
Comment (akhil-rane) [inline pkg/service/nodes.go:12]: aren't these same imports with different aliases?

Comment (shreyas-badiger) [reply pkg/service/nodes.go]: That's right. Missed this while adding the drainNode code. Thanks for catching this.

File: pkg/service/events_test.go (hunk 3 above)
Comment (shreyas-badiger) [reply pkg/service/events_test.go:56]: It is required as a part of an update to apimachinery module.

File: pkg/service/nodes.go (hunk 2 above)
Comment (shreyas-badiger) [reply pkg/service/nodes.go:84]: That's right. Modified it. Thanks for catching this!

**Review by ZihanJiang96 — APPROVED** [review]
LGTM

**Comment by figo** [comment]
not for this PR, since we have k8s client now, could we get rid of the `kubectl` in the following PR?

**Review by figo — APPROVED** [review]
LGTM

**Comment by shreyas-badiger** [comment]
Verified lifecycle-manager terminating for 100+ nodes on multiple ocassions.
**Note: All 106 instances were terminated within 5 mins.**

```
╰─ bat lcm_logs| grep "setting lifecycle event as completed with result: CONTINUE" | wc -l                                                                                                         ─╯
     106
╭─    ~ ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── ✔  17:07:32  ─╮
╰─ bat lcm_logs| grep "setting lifecycle event as completed with result: CONTINUE"                                                                                                                 ─╯
time="2023-06-29T00:01:22Z" level=info msg="i-0012f6d9cfe9556f3> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:01:23Z" level=info msg="i-099405c99fe257b6c> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:01:38Z" level=info msg="i-08a803043df1de54d> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:01:44Z" level=info msg="i-000748844ca712065> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:02:02Z" level=info msg="i-01a16fad153bb7cc6> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:02:09Z" level=info msg="i-01b4b6b6f705b9828> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:02:20Z" level=info msg="i-0395a1a75e885c20c> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:02:21Z" level=info msg="i-02b3e25305421e53d> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:02:32Z" level=info msg="i-018a63bf9c26f544c> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:03:02Z" level=info msg="i-0883ab1737f3ff702> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:03:21Z" level=info msg="i-02bc86e5df9298201> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:03:25Z" level=info msg="i-01440df2a536500ff> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:03:41Z" level=info msg="i-07190b65392bacbf6> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:03:48Z" level=info msg="i-0320f430834ef602b> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:03:51Z" level=info msg="i-02c13d60a0dd4c88d> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:03:59Z" level=info msg="i-021d38cf673553ec1> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:02Z" level=info msg="i-00d280256dd92ffe9> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:12Z" level=info msg="i-04e3dee84f076a1a3> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:13Z" level=info msg="i-06e769161945255ba> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:20Z" level=info msg="i-0a2c7f904b125d82a> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:21Z" level=info msg="i-07d66959a292d8ee9> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:23Z" level=info msg="i-03e636d60516b19cf> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:26Z" level=info msg="i-0bc42a80269d83f9f> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:28Z" level=info msg="i-037ac7aa29cf28b9f> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:29Z" level=info msg="i-06967863d304b5765> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:30Z" level=info msg="i-04c40862602879330> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:30Z" level=info msg="i-07ecc9b83db2f37c7> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:31Z" level=info msg="i-00bfee0d398d9f10f> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:31Z" level=info msg="i-0b02e3a43e39d305a> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:32Z" level=info msg="i-04a24d06da7054ee1> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:32Z" level=info msg="i-09165a253c0640e71> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:33Z" level=info msg="i-04167c8b34f37d77e> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:34Z" level=info msg="i-0c7bfc1e9062b89e4> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:35Z" level=info msg="i-0217fff187ab3134f> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:35Z" level=info msg="i-0b5853741157a9d86> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:35Z" level=info msg="i-042410623a44a52cd> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:36Z" level=info msg="i-0b4f7c73427c13a3d> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:37Z" level=info msg="i-086290754ecb3307f> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:38Z" level=info msg="i-0a65d72b415713728> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:38Z" level=info msg="i-0a91d2fc135a7f74a> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:39Z" level=info msg="i-0bbda9ead29f24231> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:39Z" level=info msg="i-03ce4c7a3e8b812ab> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:40Z" level=info msg="i-090bd87885f651ac2> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:40Z" level=info msg="i-06d0e4c64ec1b03de> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:41Z" level=info msg="i-0be3db61f3367491c> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:42Z" level=info msg="i-09775c052ffb7d853> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:42Z" level=info msg="i-0a2a9ad92e797a032> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:43Z" level=info msg="i-0ab079057933947ab> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:45Z" level=info msg="i-0ad589ee0a302b7dc> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:46Z" level=info msg="i-0c07afe33d1426d7d> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:46Z" level=info msg="i-011a8bd7f5ed1e70f> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:46Z" level=info msg="i-0bd36b8227f4f854e> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:46Z" level=info msg="i-0d513d17117c0c052> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:47Z" level=info msg="i-0ee0a2ab5e1cbbf75> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:47Z" level=info msg="i-03a029b611a55cd3b> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:47Z" level=info msg="i-0483d90112c8bde41> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:48Z" level=info msg="i-02ed7b4b3deb31860> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:48Z" level=info msg="i-07636e2671170cf24> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:48Z" level=info msg="i-0078d71b20bcebbf2> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:48Z" level=info msg="i-0fce7c07b424162b3> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:49Z" level=info msg="i-070ede9618a0ce5eb> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:49Z" level=info msg="i-04b8e0f3fee5f78a2> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:49Z" level=info msg="i-0a1e1dd8b773aca29> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:49Z" level=info msg="i-0ed9effb7a58ee6b1> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:50Z" level=info msg="i-044c173f243771517> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:50Z" level=info msg="i-01c24b237ab2ce9fc> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:51Z" level=info msg="i-0bb8d4eca2a799aff> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:52Z" level=info msg="i-0a9b7633070e91ee0> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:52Z" level=info msg="i-0b6225a50651ec2df> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:52Z" level=info msg="i-0107078fc40b28eec> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:53Z" level=info msg="i-03c199bc1852028b3> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:54Z" level=info msg="i-019b6af7b3d11ef29> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:54Z" level=info msg="i-0b57c3105de5a4bb2> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:54Z" level=info msg="i-0d654016b927d3872> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:55Z" level=info msg="i-09279ecdf8781b04a> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:55Z" level=info msg="i-077e912ea7444469e> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:55Z" level=info msg="i-0ca7de5e59d503084> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:56Z" level=info msg="i-06274b3014c14f7b5> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:56Z" level=info msg="i-0f8f3b5a9e1b65eb3> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:56Z" level=info msg="i-09058edfc7b4d0d46> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:57Z" level=info msg="i-0d7fefa867276bb18> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:57Z" level=info msg="i-0d054c777a207ee67> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:58Z" level=info msg="i-0edaceada17341840> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:04:58Z" level=info msg="i-0e77d42ba53761dba> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:00Z" level=info msg="i-0c326eeb8d4495bc4> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:01Z" level=info msg="i-0ad881c2b4b69ed45> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:02Z" level=info msg="i-0750629a602207475> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:03Z" level=info msg="i-0b8b94652a48f3c18> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:03Z" level=info msg="i-0bc76653c113f9f6c> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:04Z" level=info msg="i-0fb8a5c827edc28ab> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:04Z" level=info msg="i-08900633bddada0ad> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:05Z" level=info msg="i-0e7eb08331c8e2cdc> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:06Z" level=info msg="i-0c10302e66af669e0> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:07Z" level=info msg="i-06c6595a869553e68> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:09Z" level=info msg="i-04f6d8b5c0b8af907> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:10Z" level=info msg="i-00d083c107e2f33cc> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:11Z" level=info msg="i-0e21d4a4a668102c7> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:12Z" level=info msg="i-07072902593f6f52f> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:13Z" level=info msg="i-0c8f859b468d0fa28> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:13Z" level=info msg="i-07c9d3e8ed85378d3> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:14Z" level=info msg="i-0ef773684095d4b76> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:14Z" level=info msg="i-0e6b3de9570afc026> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:18Z" level=info msg="i-0dc9a3fd769cab2bb> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:26Z" level=info msg="i-0fb681ef3d6792747> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:31Z" level=info msg="i-0748b88f4dc4bfaf1> setting lifecycle event as completed with result: CONTINUE"
time="2023-06-29T00:05:35Z" level=info msg="i-0ccd62c96edba78c5> setting lifecycle event as completed with result: CONTINUE"
```
